File: 28-find-the-index-of-the-first-occurrence-in-a-string/find-the-index-of-the-first-occurrence-in-a-string.java
class Solution {
    public int strStr(String haystack, String needle) {
        if(needle.isEmpty()) {
            return 0;
        }
        return haystack.indexOf(needle);
    }
}
/**
class Solution{
    public int strStr(String haystack, String needle) {
        if(needle.isEmpty()){
            return 0;
        }

        int haystackLength = haystack.length();
        int needleLength = needle.length();

        for(int i=0; i <=hayStackLength - needleLength; i++){
            int j;
            for(j=0; j < needleLength; j++) {
                if(hayStack.charAt(i+j) !=needle.charAt(j)){
                    break;
                }
            }
            if(j == needleLength){
                return i;
            }
        }
        return -1;
    }
}
 */

File: 28-find-the-index-of-the-first-occurrence-in-a-string/find-the-index-of-the-first-occurrence-in-a-string.py
class Solution(object):
    def strStr(self, haystack, needle):
        """
        :type haystack: str
        :type needle: str
        :rtype: int
        """
        return haystack.find(needle)


"""
class Solution(object):
    def strStr(self, haystack, needle):
        if not needle:
            return 0;
        
        haystack_length = len(haystack)
        needle_length = len(needle)

        for i in range(haystack_length - needle_length + 1):
            j = 0
            while j < needle_length and haystack[i+j] === needle[j]:
                j+=1
            if j == needle_length:
                return i
        
        return -1
"""